compiler/rustc_parse/src/parser/nonterminal.rs

```diff
@@ -96,75 +96,76 @@ impl<'a> Parser<'a> {
                 None => {
                     return Err(self.struct_span_err(self.token.span, "expected a statement"));
                 }
             },
             NonterminalKind::PatParam { .. } | NonterminalKind::PatWithOr { .. } => {
-                token::NtPat(self.collect_tokens_no_attrs(|this| match kind {
+                Nt(token::NtPat(self.collect_tokens_no_attrs(|this| match kind {
                     NonterminalKind::PatParam { .. } => this.parse_pat_no_top_alt(None),
                     NonterminalKind::PatWithOr { .. } => {
                         this.parse_pat_allow_top_alt(None, RecoverComma::No, RecoverColon::No, CommaRecoveryMode::EitherTupleOrPipe)
                     }
                     _ => unreachable!(),
-                })?)
+                })?))
             }
 
-            NonterminalKind::Expr => token::NtExpr(self.parse_expr_force_collect()?),
+            NonterminalKind::Expr => Nt(token::NtExpr(self.parse_expr_force_collect()?)),
             NonterminalKind::Literal => {
                 // The `:literal` matcher does not support attributes
-                token::NtLiteral(
+                Nt(token::NtLiteral(
                     self.collect_tokens_no_attrs(|this| this.parse_literal_maybe_minus())?,
-                )
+                ))
             }
 
             NonterminalKind::Ty => {
-                token::NtTy(self.collect_tokens_no_attrs(|this| this.parse_no_question_mark_recover())?)
+                Nt(token::NtTy(self.collect_tokens_no_attrs(|this| this.parse_no_question_mark_recover())?))
             }
             // this could be handled like a token, since it is one
             NonterminalKind::Ident
                 if let Some((ident, is_raw)) = get_macro_ident(&self.token) =>
             {
                 self.bump();
-                token::NtIdent(ident, is_raw)
+                Nt(token::NtIdent(ident, is_raw))
             }
             NonterminalKind::Ident => {
                 let token_str = pprust::token_to_string(&self.token);
                 let msg = &format!("expected ident, found {}", &token_str);
                 return Err(self.struct_span_err(self.token.span, msg));
             }
-            NonterminalKind::Path => token::NtPath(
-                self.collect_tokens_no_attrs(|this| this.parse_path(PathStyle::Type))?,
+            NonterminalKind::Path => Nt(token::NtPath(
+                self.collect_tokens_no_attrs(|this| this.parse_path(PathStyle::Type))?),
             ),
-            NonterminalKind::Meta => token::NtMeta(P(self.parse_attr_item(true)?)),
-            NonterminalKind::TT => token::NtTT(self.parse_token_tree()),
-            NonterminalKind::Vis => token::NtVis(
-                self.collect_tokens_no_attrs(|this| this.parse_visibility(FollowedByType::Yes))?,
+            NonterminalKind::Meta => Nt(token::NtMeta(P(self.parse_attr_item(true)?))),
+            NonterminalKind::TT => Tt(self.parse_token_tree()),
```

Comment:
```suggestion
            NonterminalKind::TT => return Ok(NtOrTt::Tt(self.parse_token_tree())),
```
Then all other arms won't need to be wrapped into `Nt(...)`s.
(And `NtOrTt::*` will no longer be needed.)

Reply:
Done!